frontend/src/pages/auth/auth.tsx

```diff
@@ -21,45 +23,64 @@ const Auth: React.FC = () => {
   const dispatch = useAppDispatch();
   const { navigateAuthUser } = useAuthNavigate();
 
+  const serverError = useAuthServerError();
+
   const location: Location = useLocation();
-  const mode = location.state as ValueOf<typeof AuthMode>;
 
   const handleSignInSubmit = useCallback(
     (payload: UserSignInRequestDto): void => {
       void dispatch(authActions.signIn(payload))
         .unwrap()
-        .then((user) => {
-          navigateAuthUser(user);
-        });
+        .then((user) => navigateAuthUser(user));
     },
```

Comment:
you dont' need .uwrap.then
get user from the store, and if it's not null, then use navigate

Reply:
try something like this

 const user = useAuthUser();
 const hasUser = Boolean(user);


 if (hasUser) {
   navigateAuthUser(user));
  }